packages/@atjson/document/src/index.ts

```diff
@@ -65,19 +65,48 @@ export {
   UnknownAnnotation
 };
 
+export function getConverterFor(from: typeof Document, to: typeof Document): never | ((doc: Document) => Document) {
+  let exports = (typeof window !== 'undefined' ? window : global) as any;
+
+  let converters = exports.__atjson_converters__;
+  let converter = converters ?
+    converters[from.contentType] ?
+      converters[from.contentType][to.contentType] :
+    null :
+  null;
+  if (converter == null) {
+    throw new Error(`🚨 There is no converter registered between ${from.name} and ${to.name}.\n\nRegister a converter for this:\n\n${from.name}.defineConverterTo(${to.name}, doc => {\n  // ❤️ Write your converter here!\n  return doc;\n});`);
+  }
+
+  return converter;
+}
+
 export default class Document {
   static contentType: string;
   static schema: Array<AnnotationConstructor<any, any>> = [];
+  static debug: boolean = false;
 
   static defineConverterTo(to: typeof Document, converter: (doc: Document) => Document) {
-    if (!this.converters.has(this)) {
-      this.converters = new WeakMap();
-      this.converters.set(this, doc => doc);
+    // We may have multiple / conflicting versions of
+    // @atjson/document. To allow this, we need to
+    // register converters on the global to ensure
+    // that they can be shared across versions of the library.
+    let exports = (typeof window !== 'undefined' ? window : global) as any;
+
+    let converters = exports.__atjson_converters__;
+    if (converters == null) {
+      converters = exports.__atjson_converters__ = {};
     }
-    this.converters.set(to, converter);
-  }
 
-  private static converters: WeakMap<typeof Document, (doc: Document) => Document> = new WeakMap();
+    if (converters[this.contentType] == null) {
+      converters[this.contentType] = {};
+    }
+
+    if (!(to instanceof Document) && this.debug) {
+      console.debug(`📦 We've detected that you have multiple versions of \`@atjson/document\` installed— ${to.name} doesn't extend the same Document class as ${this.name}. If this intentional, carry on! Otherwise, you may need to dig in to see if there's any version mismatches.`);
```

Comment:
I'm _slightly_ tempted to suggest that we should just fail hard in all cases where this is true.

The obvious downside of this is if multiple dependencies have multiple versions of atjson installed, it's not going to work. I wonder if a module-local registry, rather than a global one, would help with resolving that particular issue, though?

Reply:
yeah, this may prove to be super difficult

Reply:
I'm changing this to throw an error, and to provide some pointers in terms of what went wrong